pkg/pool-weighted/contracts/smart/ManagedPoolFactory.sol

```diff
@@ -31,38 +35,40 @@ contract ManagedPoolFactory is BasePoolSplitCodeFactory, FactoryWidePauseWindow
      * @dev Deploys a new `ManagedPool`.
      */
     function create(
-        string memory name,
-        string memory symbol,
-        IERC20[] memory tokens,
-        uint256[] memory weights,
-        address[] memory assetManagers,
-        uint256 swapFeePercentage,
-        address owner,
-        bool swapEnabledOnStart,
-        bool mustAllowlistLPs,
-        uint256 managementSwapFeePercentage
-    ) external returns (address) {
+        ManagedPool.NewPoolParams calldata poolParams,
+        BasePoolController.BasePoolRights calldata basePoolRights,
+        ManagedPoolController.ManagedPoolRights calldata managedPoolRights,
+        uint256 minWeightChangeDuration
+    ) external returns (address pool) {
         (uint256 pauseWindowDuration, uint256 bufferPeriodDuration) = getPauseConfiguration();
 
-        return
-            _create(
-                abi.encode(
-                    ManagedPool.NewPoolParams({
-                        vault: getVault(),
-                        name: name,
-                        symbol: symbol,
-                        tokens: tokens,
-                        normalizedWeights: weights,
-                        assetManagers: assetManagers,
-                        swapFeePercentage: swapFeePercentage,
-                        pauseWindowDuration: pauseWindowDuration,
-                        bufferPeriodDuration: bufferPeriodDuration,
-                        owner: owner,
-                        swapEnabledOnStart: swapEnabledOnStart,
-                        mustAllowlistLPs: mustAllowlistLPs,
-                        managementSwapFeePercentage: managementSwapFeePercentage
-                    })
-                )
-            );
+        ManagedPoolController poolController = new ManagedPoolController(
+            basePoolRights,
+            managedPoolRights,
+            minWeightChangeDuration,
+            msg.sender
+        );
+
+        pool = _create(
+            abi.encode(
+                ManagedPool.NewPoolParams({
+                    vault: getVault(),
+                    name: poolParams.name,
+                    symbol: poolParams.symbol,
+                    tokens: poolParams.tokens,
+                    normalizedWeights: poolParams.normalizedWeights,
+                    assetManagers: poolParams.assetManagers,
+                    swapFeePercentage: poolParams.swapFeePercentage,
+                    pauseWindowDuration: pauseWindowDuration,
+                    bufferPeriodDuration: bufferPeriodDuration,
+                    owner: address(poolController),
+                    swapEnabledOnStart: poolParams.swapEnabledOnStart,
+                    mustAllowlistLPs: poolParams.mustAllowlistLPs,
+                    managementSwapFeePercentage: poolParams.managementSwapFeePercentage
+                })
+            )
+        );
```

Comment:
What do you think of this factory just calling into a base `ManagedPoolFactory` instead of deploying the pool directly?

Reply:
I thought we specifically *weren't* allowing that. Since we're removing all restrictions from Managed Pools, "naked" ones would be very dangerous. The ManagedPoolController is very configurable, and is intended to be suitable for 90% of the use cases (like the V1 CRP was).

If somebody really needed different functionality, they would make their own controller, but this way the UI would know that anything deployed from this factory used an approved controller, and is safe to show on the UI.

Reply:
As we're allowing asset managers on this factory they're equally as dangerous afaik and they'll all need to be individually whitelisted.

My preference for having a single "naked" `ManagedPoolFactory` that all "pool controller factories" call into comes from:
1. In the case of needing another pool controller factory then we can point it at the same underlying `ManagedPoolFactory` and know that the only changes are in the controller rather than the underlying pool
2. When tracking pools we can have an easy way of collecting all `ManagedPools` regardless of the original source by looking at the base `ManagedPoolFactory`.

Each pool controller factory would maintain a record of which pools it has deployed as well so we can keep track of which pool is from which.